superset-frontend/src/explore/actions/hydrateExplore.ts

```diff
@@ -97,6 +98,14 @@ export const hydrateExplore =
     }
 
     const initialDatasource = dataset;
+    initialDatasource.currency_formats = Object.fromEntries(
+      (initialDatasource.metrics ?? [])
+        .filter(metric => metric.currency !== undefined)
+        .map((metric): [string, Currency] => [
+          metric.metric_name,
+          metric.currency!,
```

Comment:
I think the reason why typescript may have been complaining is because the value could still be null.